prov/util/src/util_mem_monitor.c

```diff
@@ -78,10 +79,53 @@ static size_t ofi_default_cache_size(void)
 	return cache_size;
 }
 
+static int ofi_monitors_update(struct ofi_mem_monitor **monitors)
```

Comment:
Please add a comment above this function describing the synchronization problem and why we need a separate state lock.  I don't expect future readers of the code to easily understand the issue.

Reply:
Done